cloudray@cloudray-VirtualBox:~$ cd 
.cache/    .dbus/     Downloads/ .local/    Pictures/  .ssh/      
.cmake/    Desktop/   .gnupg/    .mozilla/  Public/    Templates/ 
.config/   Documents/ hadoop/    Music/     snap/      Videos/    
cloudray@cloudray-VirtualBox:~$ cd hadoop/
cloudray@cloudray-VirtualBox:~/hadoop$ cd
cd                 cd-fix-profile     cd-it8             
cd-create-profile  cd-iccdump         
cloudray@cloudray-VirtualBox:~/hadoop$ cd
cd                 cd-fix-profile     cd-it8             
cd-create-profile  cd-iccdump         
cloudray@cloudray-VirtualBox:~/hadoop$ cd hadoop-3.2.1/
cloudray@cloudray-VirtualBox:~/hadoop/hadoop-3.2.1$ bin/hadoop com.sun.tools.javac.Main Q1.java
/home/cloudray/hadoop/hadoop-3.2.1/bin/../libexec/hadoop-functions.sh: line 2366: HADOOP_COM.SUN.TOOLS.JAVAC.MAIN_USER: bad substitution
/home/cloudray/hadoop/hadoop-3.2.1/bin/../libexec/hadoop-functions.sh: line 2461: HADOOP_COM.SUN.TOOLS.JAVAC.MAIN_OPTS: bad substitution
cloudray@cloudray-VirtualBox:~/hadoop/hadoop-3.2.1$ jar cf Q1.jar Q1*.class Coordinates.class
cloudray@cloudray-VirtualBox:~/hadoop/hadoop-3.2.1$ bin/hadoop jar Q1.jar inputQ1 outputq1
Exception in thread "main" java.lang.ClassNotFoundException: inputQ1
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:316)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:236)
cloudray@cloudray-VirtualBox:~/hadoop/hadoop-3.2.1$ bin/hadoop jar Q1.jar Q1 inputQ1 outputQ1
2019-11-30 12:23:06,865 INFO impl.MetricsConfig: Loaded properties from hadoop-metrics2.properties
2019-11-30 12:23:07,145 INFO impl.MetricsSystemImpl: Scheduled Metric snapshot period at 10 second(s).
2019-11-30 12:23:07,145 INFO impl.MetricsSystemImpl: JobTracker metrics system started
Exception in thread "main" java.net.ConnectException: Call From cloudray-VirtualBox/127.0.1.1 to localhost:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:833)
	at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:757)
	at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1549)
	at org.apache.hadoop.ipc.Client.call(Client.java:1491)
	at org.apache.hadoop.ipc.Client.call(Client.java:1388)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:233)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:118)
	at com.sun.proxy.$Proxy9.getFileInfo(Unknown Source)
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:907)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
	at com.sun.proxy.$Proxy10.getFileInfo(Unknown Source)
	at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1666)
	at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1576)
	at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1573)
	at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
	at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1588)
	at org.apache.hadoop.fs.FileSystem.exists(FileSystem.java:1683)
	at org.apache.hadoop.mapreduce.lib.output.FileOutputFormat.checkOutputSpecs(FileOutputFormat.java:163)
	at org.apache.hadoop.mapreduce.JobSubmitter.checkSpecs(JobSubmitter.java:277)
	at org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:143)
	at org.apache.hadoop.mapreduce.Job$11.run(Job.java:1570)
	at org.apache.hadoop.mapreduce.Job$11.run(Job.java:1567)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1730)
	at org.apache.hadoop.mapreduce.Job.submit(Job.java:1567)
	at org.apache.hadoop.mapreduce.Job.waitForCompletion(Job.java:1588)
	at Q1.main(Q1.java:122)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:323)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:236)
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206)
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:533)
	at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:700)
	at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:804)
	at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:421)
	at org.apache.hadoop.ipc.Client.getConnection(Client.java:1606)
	at org.apache.hadoop.ipc.Client.call(Client.java:1435)
	... 38 more
cloudray@cloudray-VirtualBox:~/hadoop/hadoop-3.2.1$ jsp

Command 'jsp' not found, did you mean:

  command 'asp' from deb asp
  command 'jsr' from deb can-utils
  command 'jps' from deb openjdk-11-jdk-headless
  command 'jps' from deb openjdk-8-jdk-headless
  command 'fsp' from deb alliance
  command 'js' from deb nodejs
  command 'js' from deb rhino
  command 'jp' from deb sat-xmpp-jp
  command 'xsp' from deb mono-xsp
  command 'tsp' from deb task-spooler
  command 'jspy' from deb can-utils
  command 'jsc' from deb libjavascriptcoregtk-3.0-bin
  command 'jsc' from deb libjavascriptcoregtk-4.0-bin

Try: sudo apt install <deb name>

cloudray@cloudray-VirtualBox:~/hadoop/hadoop-3.2.1$ jps
3190 Jps

